### ISSUE TITLE
Non-standard index in Regex.Match when using Regex.replace

### ISSUE THREAD

**Issue by lazamar** [elm-lang/core#748]
Matches from Regex.replace have the value `<internal structure>` in their `index` key instead of an integer. This value always yields false when compared to other integers and returns `NaN` when summed to an integer.

This also makes it not possible to compare matches from `Regex.find` with matches from `Regex.replace` as the comparison always outputs false.

### Incompatible matches
```elm
import Html exposing (text)
import Regex exposing (..)

t = "word word word"

r = regex "\\s\\w+"

matches = find All r t


main =
  text
    <| replace All r (\m -> if List.member m matches then " y" else " n") t

-- Output : word n n
-- Expected : word y y
```

### NaN result in integer operations

Using the same values form `r`  and `t`.

```elm
replace All r (\m -> toString <| m.index + 0) t

-- Output : wordNaNNaN
-- Expected: word49
```

### Always false when compared to other ints

```elm
replace All r (\m -> toString <| List.member m.index [0..100]) t

-- Output: wordFalseFalse
-- Expected: wordTrueTrue
```

### Non-standard index in Regex.Match

```elm
replace All (regex ".") (\m -> toString m) "a"

-- Output: { match = "a", submatches = [], index = <internal structure>, number = 1 }
-- Expected: { match = "a", submatches = [], index = 0, number = 1 }
```

**Comment by rgrempel** [comment]
This may have been fixed by #736.

**Comment by lazamar** [comment]
Yes, that is it. Thanks.
